src/commons/audit.ts

```diff
@@ -1,27 +1,64 @@
 import { SQSClient, SendMessageCommand, SendMessageCommandOutput } from "@aws-sdk/client-sqs";
-import { IdentityRecordInvalidatedEvent } from "./audit-events";
+import { TxmaEvent, TxmaSisIdentityRecordInvalidated, TxmaSisStoredIdentityReadEvent } from "./audit-events";
 
 const sqsClient = new SQSClient({});
 
-const sendAuditMessage = async <T extends object>(message: T) =>
+export const sendAuditMessage = async <
+  EventName extends string,
+  ExtensionsT extends object | undefined = undefined,
+  RestrictedT extends object | undefined = undefined,
+>(
+  message: TxmaEvent<EventName, ExtensionsT, RestrictedT>
+): Promise<SendMessageCommandOutput> =>
   await sqsClient.send(
     new SendMessageCommand({
       QueueUrl: process.env.SQS_AUDIT_EVENT_QUEUE_URL,
       MessageBody: JSON.stringify(message),
     })
   );
 
+const createDefaultEventFields = <EventName extends string>(
+  eventName: EventName,
+  userId: string,
+  govukSigninJourneyId?: string
+): TxmaEvent<EventName, any, any> => {
+  return {
+    component_id: process.env.COMPONENT_ID,
+    event_name: eventName,
+    event_timestamp_ms: Date.now(),
+    timestamp: Math.floor(Date.now() / 1000),
+    user: {
+      user_id: userId,
+      ...(govukSigninJourneyId ? { govuk_signin_journey_id: govukSigninJourneyId } : {}),
+    },
+    extensions: {},
```

Comment:
I recall a thing coming up, in recent history, that if an event had no `extensions` or `restricted` entries, then we should not send empty objects.

Including these in the defaults could mean we end up doing this.